test/data_item_test.dart

```diff
@@ -0,0 +1,217 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:arweave/src/streams/data_item.dart';
+import 'package:arweave/src/streams/data_models.dart';
+import 'package:arweave/src/streams/errors.dart';
+import 'package:arweave/src/streams/utils.dart';
+import 'package:fpdart/fpdart.dart';
+import 'package:test/test.dart';
+import 'fixtures/test_wallet.dart';
+import 'deserialize_tags.dart' if (dart.library.io) 'deserialize_tags_io.dart';
+
+void main() async {
+  final wallet = getTestWallet();
+  final walletOwner = await wallet.getOwner();
+
+  final dataText = 'Hello World!';
+
+  final data = utf8.encode(dataText);
+
+  dataStreamGenerator() {
+    return Stream.fromIterable([data]).map((list) => Uint8List.fromList(list));
+  }
+
+  final tags = [
+    createTag('First-Tag', 'First-Value'),
+    createTag('Second-Tag', 'Second-Value'),
+  ];
+
+  const expectedDataItemId = 'PKvhRDCiv_gpusxagkfTdjWJrp1RazFuxl04E6FUQqs';
+  const expectedDataItemSize = 1104;
+
+  group('[streams][data_item]', () {
+    test('create and sign data item', () async {
```

Comment:
Nit: we usually set the top-level group's description as the name of the component we're testing so you can read: "Component X behaves like THIS with the input A" - E.g. "createDataItemTaskEither method" "instantiates a stream that creates and sign a data item"